bin/veritech/src/main.rs

```diff
@@ -49,13 +49,20 @@ async fn async_main() -> Result<()> {
             .set_verbosity_and_wait(args.verbose.into())
             .await?;
     }
+    let graceful_shutdown_timeout = match args.graceful_shutdown_timeout_secs {
+        Some(provided) => Duration::from_secs(provided),
+        None => DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT,
+    };
     debug!(arguments =?args, "parsed cli arguments");
 
     let config = Config::try_from(args)?;
     debug!(?config, "computed configuration");
 
-    let server = Server::from_config(config, main_token.clone()).await?;
+    let (server, maybe_heartbeat_app) = Server::from_config(config, main_token.clone()).await?;
 
+    if let Some(mut heartbeat_app) = maybe_heartbeat_app {
+        main_tracker.spawn(async move { heartbeat_app.run().await });
+    }
```

Comment:
The run method borrows `self` mutably. Moving the app using `async move` is easy and mimics how the main server runs.

Also... we probably can run the app in the main server itself, but that is for another day.